lib/jxl/modular/transform/enc_palette.cc

```diff
@@ -320,6 +321,20 @@ Status FwdPaletteIteration(Image &input, uint32_t begin_c, uint32_t end_c,
     }
   }
 
+  std::map<std::vector<pixel_type>, bool> implicit_color;
```

Comment:
We really should write a simple hashtable implementation and/or a "small vector" for these things... (not now ;))

Reply:
Agreed, and possibly also special-case anything up to 4 channels uint16 to do something like e1/fjxl where it takes interleaved uint8/uint16 buffers directly as input (instead of this very generic code we have here now). It's rather wasteful to convert interleaved ints to planar floats and then planar ints and then do palette :)
Also agreed not now :)